### PR TITLE
geyser: wrap messages in Arc

### PR DESCRIPTION
**Pull request by fanatid**

### CODE REVIEW

File: yellowstone-grpc-proto/src/weak/filter.rs [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+//             message.data.clone()
+//         } else {
+//             let mut data =
+//                 Vec::with_capacity(accounts_data_slice.iter().map(|s| s.end - s.start).sum());
+//             for slice in accounts_data_slice {
+//                 if message.data.len() >= slice.end {
+//                     data.extend_from_slice(&message.data[slice.start..slice.end]);
+//                 }
+//             }
+//             data
```
Comment (lvboudre) [inline yellowstone-grpc-proto/src/weak/filter.rs]: Dead code to remove?

**Comment by fanatid** [comment]
benchmark:
```
accounts/ref            time:   [14.624 ms 15.003 ms 15.448 ms]
                        change: [-4.6751% -0.8189% +2.9899%] (p = 0.69 > 0.05)
                        No change in performance detected.
Found 14 outliers among 100 measurements (14.00%)
  6 (6.00%) high mild
  8 (8.00%) high severe

accounts/prost          time:   [26.237 ms 26.899 ms 27.687 ms]
                        change: [-6.0325% -2.4645% +1.6056%] (p = 0.21 > 0.05)
                        No change in performance detected.
Found 14 outliers among 100 measurements (14.00%)
  7 (7.00%) high mild
  7 (7.00%) high severe

accounts/prost clone    time:   [33.759 ms 34.480 ms 35.326 ms]
Found 10 outliers among 100 measurements (10.00%)
  2 (2.00%) high mild
  8 (8.00%) high severe

transactions/ref        time:   [84.448 µs 86.141 µs 88.175 µs]
                        change: [-5.9081% -3.3280% -0.8504%] (p = 0.01 < 0.05)
                        Change within noise threshold.
Found 20 outliers among 100 measurements (20.00%)
  6 (6.00%) low mild
  4 (4.00%) high mild
  10 (10.00%) high severe

transactions/prost      time:   [220.35 µs 225.02 µs 230.68 µs]
                        change: [-2.3180% +1.5346% +5.6245%] (p = 0.45 > 0.05)
                        No change in performance detected.
Found 12 outliers among 100 measurements (12.00%)
  2 (2.00%) low mild
  1 (1.00%) high mild
  9 (9.00%) high severe

transactions/prost clone
                        time:   [226.24 µs 231.22 µs 237.29 µs]
                        change: [-1.2639% +1.6528% +4.8923%] (p = 0.31 > 0.05)
                        No change in performance detected.
Found 10 outliers among 100 measurements (10.00%)
  1 (1.00%) low mild
  9 (9.00%) high severe

Benchmarking blocks/ref: Warming up for 3.0000 s
Warning: Unable to complete 100 samples in 5.0s. You may wish to increase target time to 15.7s, or reduce sample count to 30.
blocks/ref              time:   [153.60 ms 155.80 ms 158.21 ms]
                        change: [-1.7113% +0.3508% +2.3395%] (p = 0.75 > 0.05)
                        No change in performance detected.
Found 19 outliers among 100 measurements (19.00%)
  1 (1.00%) high mild
  18 (18.00%) high severe

Benchmarking blocks/prost: Warming up for 3.0000 s
Warning: Unable to complete 100 samples in 5.0s. You may wish to increase target time to 30.8s, or reduce sample count to 10.
blocks/prost            time:   [305.61 ms 308.53 ms 311.57 ms]
                        change: [-3.8054% -2.5197% -1.2056%] (p = 0.00 < 0.05)
                        Performance has improved.

Benchmarking blocks/prost clone: Warming up for 3.0000 s
Warning: Unable to complete 100 samples in 5.0s. You may wish to increase target time to 29.3s, or reduce sample count to 10.
blocks/prost clone      time:   [291.08 ms 293.96 ms 297.00 ms]
                        change: [-1.9567% -0.5771% +0.8371%] (p = 0.43 > 0.05)
                        No change in performance detected.
```

**Comment by fanatid** [comment]
> I don't see unit test for the new code.

do you mean encoding? tests in `message_ref.rs`

> In fumarole I use yellowstone-grpc-geyser and proto, will this changes break my code?

proto crate should not introduce any breaking changes, only new features
geyser crate is not for public usage, what do you use?

**Comment by lvboudre** [comment]
> > I don't see unit test for the new code.
> 
> do you mean encoding? tests in `message_ref.rs`
> 
> > In fumarole I use yellowstone-grpc-geyser and proto, will this changes break my code?
> 
> proto crate should not introduce any breaking changes, only new features geyser crate is not for public usage, what do you use?

My bad, github collapsed your message_ref.

**Pull request closed without merging**